plugins/parsers/json/parser.go

```diff
@@ -57,19 +57,59 @@ func (p *JSONParser) parseObject(metrics []telegraf.Metric, jsonOut map[string]i
 	}
 
 	f := JSONFlattener{}
-	err := f.FlattenJSON("", jsonOut)
+	err := f.FullFlattenJSON("", jsonOut, true, true)
 	if err != nil {
 		return nil, err
 	}
 
-	metric, err := metric.New(p.MetricName, tags, f.Fields, time.Now().UTC())
+	tags, nFields := p.switchFieldToTag(tags, f.Fields)
+
+	metric, err := metric.New(p.MetricName, tags, nFields, time.Now().UTC())
 
 	if err != nil {
 		return nil, err
 	}
 	return append(metrics, metric), nil
 }
 
+//will take in field map with strings and bools,
+//search for TagKeys that match fieldnames and add them to tags
+//will delete any strings/bools that shouldn't be fields
+//assumes that any non-numeric values in TagKeys should be displayed as tags
+func (p *JSONParser) switchFieldToTag(tags map[string]string, fields map[string]interface{}) (map[string]string, map[string]interface{}) {
+	for _, name := range p.TagKeys {
+		//switch any fields in tagkeys into tags
+		if fields[name] != nil {
+			switch fields[name].(type) {
+			//case of field is string
+			case string:
+				tags[name] = fields[name].(string)
+				delete(fields, name)
+			case bool:
+				tags[name] = strconv.FormatBool(fields[name].(bool))
+				delete(fields, name)
+			case int:
```

Comment:
I think it is not possible to have ints in parsed JSON, so we can remove this case.  Let's also add a warning though if we hit the default case:
```
log.Printf("E! [parsers.json] Unrecognized type %T", value)
```